doozer/doozerlib/metadata.py

```diff
@@ -401,11 +402,117 @@ async def get_latest_build(self, **kwargs):
+        Under 'is' for RPMs, we expect 'el7' and/or 'el8', etc. For images, just 'nvr'.
+        """
+
+        is_config = self.config['is']
+
+        if self.meta_type == 'rpm':
+            if el_target is None:
+                raise ValueError(
+                    f'Expected el_target to be set when querying a pinned RPM component {self.distgit_key}')
+            is_nvr = is_config[f'el{el_target}']
+            if not is_nvr:
+                self.logger.warning('No pinned NVR found for RPM %s and target %s', self.rpm_name, el_target)
+                return None
+
+        else:
+            # The image metadata (or, more likely, the current assembly) has the image
+            # pinned. Return only the pinned NVR. When a child image is being rebased,
+            # it uses get_latest_build to find the parent NVR to use (if it is not
+            # included in the "-i" doozer argument). We need it to find the pinned NVR
+            # to place in its Dockerfile.
+            # Pinning also informs gen-payload when attempting to assemble a release.
+            is_nvr = is_config.nvr
+            if not is_nvr:
+                raise ValueError(f'Did not find nvr field in pinned Image component {self.distgit_key}')
+
+        # strict means raise an exception if not found.
+        return await self.runtime.konflux_db.get_build_record_by_nvr(is_nvr, strict=True)
+
+    async def get_latest_konflux_build(self, assembly: Optional[str] = None,
+                                       outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS,
+                                       el_target: Optional[int] = None,
+                                       honor_is: bool = True,
+                                       completed_before: Optional[datetime] = None,
+                                       extra_patterns: dict = {},
+                                       **kwargs) -> Optional[KonfluxBuildRecord]:
+        """
+        :param assembly: A non-default assembly name to search relative to. If not specified, runtime.assembly
+                         will be used. If runtime.assembly is also None, the search will return true latest.
+                         If the assembly parameter is set to '', this search will also return true latest.
+        :param outcome: one in KonfluxBuildOutcome[FAILURE, SUCCESS, PENDING]
+        :param el_target: specify 'el7', 'el8', etc.
+                          For an RPM, leave as None if you want the true latest.
+                          For an image, leaving as none will default to the RHEL version in the branch
+                          associated with the metadata.
+        :param honor_is: If True, and an assembly component specifies 'is', that nvr will be returned.
+        :param completed_before: cut off timestamp for builds completion time
+        :param extra_patterns: e.g. {'release': 'b45ea65'} will result in adding "AND release LIKE '%b45ea65%'" to the query
+        """
+
```

Comment:
I'm looking at the code for `get_latest_brew_build` 
https://github.com/openshift-eng/art-tools/blob/de3d8b791c0bf29f012f103e6d4671e0ff29d520/doozer/doozerlib/metadata.py#L412
We have params that are not used in this func, like component_name. Now that it's irrelevant, maybe we have a check that fails if the component name is different from what we determine

Reply:
`component_name` is used by `get_latest_brew_build` to get the package ID, and this is later used to fetch the builds:

```
builds = koji_api.listBuilds(packageID=package_id,
```
Overriding it to match something different from the metadata component name could be used in particular cases (e.g. olm bundles?) although I haven't found any occurrences of  this. Either way, I feel this and other refactoring changes should be postponed as things are getting messy already and I want to limit changes to what's strictly needed for this task